src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift

```diff
@@ -338,7 +339,30 @@ class ExposureSubmissionCoordinator: NSObject, ExposureSubmissionCoordinating, R
 			onPrimaryButtonTap: { [weak self] isLoading in
 				self?.model.exposureSubmissionService.acceptPairing()
 				self?.model.exposureSubmissionService.isSubmissionConsentGiven = true
-				self?.showQRScreen(isLoading: isLoading)
+				if #available(iOS 14.4, *) {
+					self?.exposureManager.preAuthorizeKeys(completion: { error in
+						DispatchQueue.main.async { [weak self] in
+							if let error = error as? ENError {
+								switch error.toExposureSubmissionError() {
+								case .notAuthorized:
+									// user did not authorize -> continue to scanning the qr code
+									self?.showQRScreen(isLoading: isLoading)
+								default:
+									// present alert
+									let alert = UIAlertController.errorAlert(message: error.localizedDescription, completion: { [weak self] in
+										self?.showQRScreen(isLoading: isLoading)
+									})
+									self?.navigationController?.present(alert, animated: true, completion: nil)
+								}
+							} else {
+								// continue to scanning the qr code
+								self?.showQRScreen(isLoading: isLoading)
```

Comment:
The comment on line 348 is a bit confusing to me maybe you can clarify, shouldn't we request Authorization  if the user is not authorized? or do we do it **after** he scans the QR code?

Reply:
We do it after